sound/soc/codecs/rt715.c

```diff
@@ -319,7 +319,10 @@ static int rt715_mux_get(struct snd_kcontrol *kcontrol,
 	/* FIXME: PLB: check return status on read/write */
 	snd_soc_component_write(component, reg, 0x0);
 	snd_soc_component_read(component, RT715_READ_HDA_0, &val);
-	if ((e->reg == 0x24 || e->reg == 0x25) && (val > 0))
+	/* According to return value of mux of nid RT715_MUX_IN3 and
+	RT715_MUX_IN4, we should convert value here to match the amixer
+	item number. */
+	if ((e->reg == RT715_MUX_IN3 || e->reg == RT715_MUX_IN4) && (val > 0))
```

Comment:
i just don't understand the math here. this does not align with the comment below that we skip the value 1?

should this be (val == 1)?

Reply:
@plbossart  I'll send you an email to explain this if condition more detail.

Reply:
> i just don't understand the math here. this does not align with the comment below that we skip the value 1?
> 
> should this be (val == 1)?

@plbossart The main issue is that the first two index of ADC Mux 24/25 are routed to the same source. ie, ADC Mux 24 0/1 will both connect to "MIC2". Previously, the driver presents all of them to DAPM routes/controls. But it will make DAPM confused. So the patch remove index 1 of the Muxes. That introduces an issue that the Mux connection index is not match to the register value. i.e.
```
Mux index,     register index
    0                    0
    1                    2
    2                    3
    3                    4
    4                    5
```
What @jack-cy-yu want to do here is to convert the register index to mux index. So we can get the correct value from user space.

Reply:
@thanks @bardliao 
@jack-cy-yu the code is fine, but you can just use a comment such as

"the  first two indices of ADC Mux 24/25 are routed to the same hardware source. ie, ADC Mux 24 0/1 will both connect to MIC2. To have a unique set of inputs, we skip the index1 of the muxes"

Reply:
@jack-cy-yu also please fix the style/typos so that we don't have to redo this later for upstream. Thanks!